packages/transaction-controller/src/helpers/MultichainTrackingHelper.ts

```diff
@@ -451,4 +458,31 @@ export class MultichainTrackingHelper {
       this.#etherscanRemoteTransactionSourcesMap.delete(chainId);
     });
   };
+
+  #getNetworkClient({
+    networkClientId,
+    chainId,
+  }: {
+    networkClientId?: NetworkClientId;
+    chainId?: Hex;
+  } = {}): NetworkClient | undefined {
+    let networkClient: NetworkClient | undefined;
+
+    if (networkClientId) {
+      try {
+        networkClient = this.#getNetworkClientById(networkClientId);
+      } catch (err) {
+        log('failed to get network client by networkClientId');
+      }
+    }
+    if (!networkClient && chainId) {
+      try {
+        networkClientId = this.#findNetworkClientIdByChainId(chainId);
```

Comment:
I appreciate this code was just moved, but should we avoid reassigning to an argument and instead call this something like `networkClientIdForChainId` for clarity?

Reply:
Done in https://github.com/MetaMask/core/pull/4055/commits/bfe24c3f198295f4aa80b8f42bbb2e74bdc1664d